crates/vault-registry/src/lib.rs

```diff
@@ -265,19 +264,12 @@ pub mod pallet {
         /// * `public_key` - the BTC public key of the vault to update
         #[pallet::weight(<T as Config>::WeightInfo::update_public_key())]
         #[transactional]
-        pub fn update_public_key(
-            origin: OriginFor<T>,
-            currency_pair: DefaultVaultCurrencyPair<T>,
-            public_key: BtcPublicKey,
-        ) -> DispatchResultWithPostInfo {
+        pub fn update_public_key(origin: OriginFor<T>, public_key: BtcPublicKey) -> DispatchResultWithPostInfo {
```

Comment:
Not sure if we should keep this name. Maybe `register_bitcoin_public_key` would be better?

Reply:
Yeah agree :+1: or even `set_public_key` if we still want to allow changing

Reply:
Changed to set_public_key. Although I am not sure why we alllow changing bitcoin public keys, that might complicate imports on the client side..